### PR TITLE
feat(widgets): Collect iterator of `Line` into `Tabs`

### PR DESCRIPTION
**Pull request by Lunderberg**
A follow-up from https://github.com/ratatui-org/ratatui/pull/755, allowing any iterator whose item is convertible into `Line` to be collected into `Tabs`.

### CODE REVIEW

File: src/widgets/tabs.rs
Comment (joshka) [reply src/widgets/tabs.rs:99]: > Sounds good. That was something I was wondering on, but figured I'd err on the side of backwards compatibility when it isn't a project I've contributed to before.

I've been fairly OK with breaking compatibility for things that are easy to fix like type inference. This is something that generally breaks a single line of code in the user's application, and there's no good way to deprecate it first. Other places we've tried to at least give a couple of releases notification with deprecated attributes (especially for changes that affect a lot of code).

Also, I'm rarely worried about performance of copying for this library even where it isn't optimized for release builds, mainly because for normal screen sizes even bad algorithms should generally have a fairly low N (~50 rows, ~200 cols), so for the most part optimization is noise (unless proven otherwise by benchmarks).

Comment (Lunderberg) [reply src/widgets/tabs.rs]: Sounds good.  I've updated the PR to accept any `IntoIterator<Item: Into<Line>>`  for calls to `Tabs::new`, and updated the two examples that relied on the previous behavior for type inference.

**Review by joshka — APPROVED** [review]
LGTM. Needs a commit message update (for the changelog)

**Review by joshka — CHANGES_REQUESTED** [review]
Actually - the commit message also should call out the breaking change and note it in BREAKING-CHANGES.md (similar to the note about List in #672)

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
Can you please resolve the breaking changes doc conflict?
This will allow me to merge without another maintainer's approval.

**Comment by Lunderberg** [comment]
@joshka Sure!  Merge conflict is now resolved.

**Review by joshka — APPROVED** [review]

**Comment by joshka** [comment]
Modified the commit message a little and merged. Thanks for the PR :)

**Pull request merged**
